File: src/main/java/com/onmeet/common/exception/ErrorCode.java
package com.onmeet.common.exception;

import lombok.Getter;
import org.springframework.http.HttpStatus;

@Getter
public enum ErrorCode {
    INVALID_REQUEST(HttpStatus.BAD_REQUEST, "Invalid request"),
    NOT_FOUND(HttpStatus.NOT_FOUND, "Resource not found"),
    UNAUTHORIZED(HttpStatus.UNAUTHORIZED, "Unauthorized"),
    FORBIDDEN(HttpStatus.FORBIDDEN, "Forbidden"),
    INTERNAL_ERROR(HttpStatus.INTERNAL_SERVER_ERROR, "Internal server error");

    private final HttpStatus status;
    private final String message;

    ErrorCode(HttpStatus status, String message) {
        this.status = status;
        this.message = message;
    }

}

File: onmeet-common/src/main/kotlin/com/onmeet/common/exception/ErrorCode.kt
package com.onmeet.common.exception

import org.springframework.http.HttpStatus

/**
 * 모든 서비스의 에러 코드가 구현해야 하는 인터페이스.
 * 각 서비스는 이 인터페이스를 구현한 enum을 정의한다.
 */
interface ErrorCode {
    /** 고유 에러 코드 (예: AUTH_001, VIDEO_002) */
    val code: String
    /** 프론트엔드에 전달될 한글 에러 메시지 */
    val message: String
    /** HTTP 상태 코드 */
    val status: HttpStatus
}
